Src/Support/Google.Apis.Auth/OAuth2/IGoogleCredential.cs

```diff
@@ -37,5 +40,28 @@ internal interface IGoogleCredential : ICredential, ITokenAccessWithHeaders
         /// <returns>A new instance with the same type as this but with <see cref="QuotaProject"/>
         /// set to <paramref name="quotaProject"/>.</returns>
         IGoogleCredential WithQuotaProject(string quotaProject);
+
+        /// <summary>
+        /// Returns true only if this credential type has no scopes by default and requires
+        /// a call to <see cref="MaybeWithScopes(IEnumerable{string})"/> before use.
```

Comment:
Well, it doesn't necessarily *require* a call to MaybeWithScopes... service account credentials can be used with scopes, or with self-signed JWTs, for example.

Reply:
Ah yes that's true, and I'll change it here.
But that's not the semantics of GoogleCredential.IsCreateScopedRequired, which does return true when there are no scopes, even though it is really not required to call CreateScoped (I copied this doc from GoogleCredential).
Should we start returning false always on GoogleCredential.IsCreateScopedRequired?
And maybe change this one for HasScopes property instead tu surface on GoogleCredential?
I'll do that and you can take a look.